fili-core/src/main/java/com/yahoo/bard/webservice/util/StreamUtils.java

```diff
@@ -145,4 +146,137 @@
                 return castedOperand;
         };
     }
+
+    /**
+     * Build a function that can map a Map from one map to another.
+     *
+     * @param keyMapper  Function to map from the input Map's Entries to the result Map's keys
+     * @param valueMapper  Function to map from the input Map's Entries to the result Map's values
+     * @param resultSupplier  Supplier of the result map instance
+     * @param <A>  Type of the input Map's keys
+     * @param <B>  Type of the input Map's values
+     * @param <K>  Type of the result Map's keys
+     * @param <V>  Type of the result Map's values
+     * @param <R>  Type of the input Map
+     * @param <T>  Type of the result map
+     *
+     * @return a function that can map a Map from one map to another.
+     */
+    public static <A, B, K, V, R extends Map<A, B>, T extends Map<K, V>> Function<R, T> mapMap(
+            Function<? super Map.Entry<A, B>, ? extends K> keyMapper,
+            Function<? super Map.Entry<A, B>, ? extends V> valueMapper,
+            Supplier<T> resultSupplier
+    ) {
+        return (R input) -> input.entrySet().stream()
+                .collect(
+                        resultSupplier,
+                        (result, entry) -> result.put(keyMapper.apply(entry), valueMapper.apply(entry)),
+                        Map::putAll
```

Comment:
The problem with this is that if the keymapper maps keys into a smaller space, then the ordering of the entries will potentially alter the results on the mapping, somewhat unpredictably.  A more general case of this should probably take a collision managing joiner.